### PR TITLE
fix GTK client message log window does not restore level selection

### LINKED ISSUE #4181: Message log window doesn't restore level selection on first show
### What is the issue?

To reproduce:
1. Start GTK client
2. Click Help menu -> Message Log item to open the window

Expected result: Level combo contains last selected level.

Actual result: Level combo is in indeterminate state (no item is selected).

Worked correctly in 4.0.0-beta.1.

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

134363d83de13d799dacc5465fef871a3083ca6c

### LINKED ISSUE #4233: remove 'trace' option from message log UI
### What is the issue?

Reproduce:
Start GTK 3 client.
Help -> Message Log. Select level Trace. Start to download any kind of torrent. Wait about 1 or 2 mins and you'll got not responding window.

OS: Debian Bookworm x86_64. Custom compile with default flags, clang14 from "main" branch(3ecf0dfccd627c1a137bd6b5103f9190e0d96af6).
Debug/Release also not responding.

First run gdb backtrace:
```
(gdb) bt
#0  0x00007ffff70ae155 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#1  0x00007ffff70ae7a1 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#2  0x00007ffff70ae436 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#3  0x00007ffff70ae29f in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#4  0x00007ffff70b0a18 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#5  0x00007ffff70b0ddc in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#6  0x00007ffff71766e0 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#7  0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#10 0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#11 0x00007ffff7165699 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#12 0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff7161f5d in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#17 0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#18 0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#19 0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#20 0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#21 0x00007ffff704672c in gtk_list_store_set_value () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#22 0x00005555556849bf in Gtk::TreeRow::set_value<unsigned int>(Gtk::TreeModelColumn<unsigned int> const&, unsigned int const&) const (this=0x7fffffffd280, column=..., data=@0x555555b87e88: 19962) at /usr/include/gtkmm-3.0/gtkmm/treeiter.h:580
#23 0x00005555556843d8 in Gtk::TreeValueProxy<unsigned int>::operator=(unsigned int const&) (this=0x7fffffffd1f0, data=@0x555555b87e88: 19962) at /usr/include/gtkmm-3.0/gtkmm/treeiter.h:552
#24 0x00005555556ee63f in (anonymous namespace)::addMessages(Glib::RefPtr<Gtk::ListStore> const&, tr_log_message*) (store=..., head=0x555557b755e0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:366
#25 0x00005555556ee2e4 in MessageLogWindow::Impl::onRefresh() (this=0x5555564b4150) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:397
#26 0x00005555556f9412 in sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>::operator()() const (this=0x555556f375c8) at /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
#27 0x00005555556f9389 in sigc::adaptor_functor<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl> >::operator()() const (this=0x555556f375c0) at /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
#28 0x00005555556f9361 in sigc::internal::slot_call0<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>, bool>::call_it(sigc::internal::slot_rep*) (rep=0x555556f37590) at /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
#29 0x00007ffff66a5fb2 in  () at /lib/x86_64-linux-gnu/libglibmm-2.4.so.1
#30 0x00007ffff650719a in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#31 0x00007ffff650667f in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#32 0x00007ffff6506a38 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#33 0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#34 0x00007ffff67b353d in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#35 0x00005555556c78ce in main(int, char**) (argc=1, argv=0x7fffffffe008) at /home/xxx/dev/transmission/gtk/main.cc:127
(gdb) f 24
#24 0x00005555556ee63f in (anonymous namespace)::addMessages (store=..., head=0x555557b755e0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:366
366	        row[message_log_cols.sequence] = ++sequence;
(gdb) l
361	        auto row_it = store->prepend();
362	        auto& row = *row_it;
363	        row[message_log_cols.tr_msg] = &message;
364	        row[message_log_cols.name] = name;
365	        row[message_log_cols.message] = message.message;
366	        row[message_log_cols.sequence] = ++sequence;
367	
368	        /* if it's an error message, dump it to the terminal too */
369	        if (message.level == TR_LOG_ERROR)
370	        {
(gdb) info locals
message = @0x555557b75570: {level = TR_LOG_TRACE, file = "/home/xxx/dev/transmission/libtransmission/bandwidth.cc", line = 187, when = 1669274052, name = Python Exception <class 'gdb.error'>: There is no member named _M_p.
, message = Python Exception <class 'gdb.error'>: There is no member named _M_p.
, next = 0x555557b755e0}
name = 0x555557b63e80 "bandwidth.cc:187"
row_it = {<Gtk::TreeIterBase> = {<No data fields>}, model_ = 0x555556c87688, is_end_ = false}
row = @0x7fffffffd280: {<Gtk::TreeIter> = {<Gtk::TreeIterBase> = {<No data fields>}, model_ = 0x555556c87688, is_end_ = false}, <No data fields>}
sequence = 19962
default_name = <incomplete type>
```

Second try backtrace:
```
(gdb) thread apply all bt

Thread 19 (Thread 0x7fffedc706c0 (LWP 124683) "transmission-gt"):
#0  __futex_abstimed_wait_common64 (private=0, cancel=true, abstime=0x0, op=393, expected=0, futex_word=0x5555565f4c20) at ./nptl/futex-internal.c:57
#1  __futex_abstimed_wait_common (futex_word=futex_word@entry=0x5555565f4c20, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0, cancel=cancel@entry=true) at ./nptl/futex-internal.c:87
#2  0x00007ffff5bc5d9b in __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x5555565f4c20, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=private@entry=0) at ./nptl/futex-internal.c:139
#3  0x00007ffff5bc83f8 in __pthread_cond_wait_common (abstime=0x0, clockid=0, mutex=0x5555565f4bd0, cond=0x5555565f4bf8) at ./nptl/pthread_cond_wait.c:503
#4  ___pthread_cond_wait (cond=0x5555565f4bf8, mutex=0x5555565f4bd0) at ./nptl/pthread_cond_wait.c:618
#5  0x0000555555854575 in std::condition_variable::wait<tr_web::Impl::curlThreadFunc(tr_web::Impl*)::{lambda()#1}>(std::unique_lock<std::mutex>&, tr_web::Impl::curlThreadFunc(tr_web::Impl*)::{lambda()#1}) (this=0x5555565f4bf8, __lock=..., __p=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/condition_variable:102
#6  0x0000555555852f03 in tr_web::Impl::curlThreadFunc(tr_web::Impl*) (impl=0x5555565f4b40) at /home/xxx/dev/transmission/libtransmission/web.cc:497
#7  0x0000555555854285 in std::__invoke_impl<void, void (*)(tr_web::Impl*), tr_web::Impl*>(std::__invoke_other, void (*&&)(tr_web::Impl*), tr_web::Impl*&&) (__f=@0x555556cf4e50: 0x555555852db0 <tr_web::Impl::curlThreadFunc(tr_web::Impl*)>, __args=@0x555556cf4e48: 0x5555565f4b40) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:61
#8  0x00005555558541d2 in std::__invoke<void (*)(tr_web::Impl*), tr_web::Impl*>(void (*&&)(tr_web::Impl*), tr_web::Impl*&&) (__fn=@0x555556cf4e50: 0x555555852db0 <tr_web::Impl::curlThreadFunc(tr_web::Impl*)>, __args=@0x555556cf4e48: 0x5555565f4b40) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:96
#9  0x0000555555854192 in std::thread::_Invoker<std::tuple<void (*)(tr_web::Impl*), tr_web::Impl*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) (this=0x555556cf4e48) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:252
#10 0x0000555555854145 in std::thread::_Invoker<std::tuple<void (*)(tr_web::Impl*), tr_web::Impl*> >::operator()() (this=0x555556cf4e48) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:259
#11 0x0000555555854019 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(tr_web::Impl*), tr_web::Impl*> > >::_M_run() (this=0x555556cf4e40) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:210
#12 0x00007ffff58d44a3 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#13 0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#14 0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
Warning: the current language does not match this frame.

Thread 18 (Thread 0x7fffecb146c0 (LWP 124682) "transmission-gt"):
#0  0x00007ffff5c48cb6 in epoll_wait (epfd=16, events=0x555556436f00, maxevents=32, timeout=40) at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
#1  0x00007ffff63d7814 in  () at /lib/x86_64-linux-gnu/libevent-2.1.so.7
#2  0x00007ffff63cd8a1 in event_base_loop () at /lib/x86_64-linux-gnu/libevent-2.1.so.7
#3  0x0000555555941d5a in tr_session_thread_impl::sessionThreadFunc(event_base*) (this=0x55555658dad0, evbase=0x55555662c580) at /home/xxx/dev/transmission/libtransmission/session-thread.cc:244
#4  0x00005555559430e9 in std::__invoke_impl<void, void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*>(std::__invoke_memfun_deref, void (tr_session_thread_impl::*&&)(event_base*), tr_session_thread_impl*&&, event_base*&&) (__f=@0x555556807248: (void (tr_session_thread_impl::*)(tr_session_thread_impl * const, event_base *)) 0x555555941c80 <tr_session_thread_impl::sessionThreadFunc(event_base*)>, __t=@0x555556807240: 0x55555658dad0, __args=@0x555556807238: 0x55555662c580) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:74
#5  0x0000555555942fb7 in std::__invoke<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*>(void (tr_session_thread_impl::*&&)(event_base*), tr_session_thread_impl*&&, event_base*&&) (__fn=@0x555556807248: (void (tr_session_thread_impl::*)(tr_session_thread_impl * const, event_base *)) 0x555555941c80 <tr_session_thread_impl::sessionThreadFunc(event_base*)>, __args=@0x555556807240: 0x55555658dad0, __args=@0x555556807238: 0x55555662c580) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/invoke.h:96
#6  0x0000555555942f5b in std::thread::_Invoker<std::tuple<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >::_M_invoke<0ul, 1ul, 2ul>(std::_Index_tuple<0ul, 1ul, 2ul>) (this=0x555556807238) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:252
#7  0x0000555555942ef5 in std::thread::_Invoker<std::tuple<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >::operator()() (this=0x555556807238) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:259
#8  0x0000555555942cf9 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> > >::_M_run() (this=0x555556807230) at /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_thread.h:210
#9  0x00007ffff58d44a3 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#10 0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#11 0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 11 (Thread 0x7ffff01fe6c0 (LWP 124675) "dconf worker"):
#0  0x00007ffff5c3c0af in __GI___poll (fds=0x555556577e50, nfds=1, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff65069ae in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007fffee47d3bd in  () at /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
#4  0x00007ffff6530d0d in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 3 (Thread 0x7ffff35bd6c0 (LWP 124666) "gdbus"):
#0  0x00007ffff5c3c0af in __GI___poll (fds=0x5555569cee80, nfds=4, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff65069ae in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6506cef in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff67ee296 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00007ffff6530d0d in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 2 (Thread 0x7ffff3dbe6c0 (LWP 124665) "gmain"):
#0  0x00007ffff5c3c0af in __GI___poll (fds=0x5555569b1c10, nfds=2, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff65069ae in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff6506b11 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff6530d0d in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff5bc8fd4 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#6  0x00007ffff5c4966c in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81

Thread 1 (Thread 0x7ffff413f0c0 (LWP 124664) "transmission-gt"):
#0  0x00007ffff65216f3 in g_sequence_foreach_range () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#1  0x00007ffff716191e in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#2  0x00007ffff65ff3b0 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#3  0x00007ffff6612076 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#4  0x00007ffff6618bf5 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#5  0x00007ffff6618dbf in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff7046d94 in gtk_list_store_insert () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#7  0x00007ffff7b7d48a in Gtk::ListStore::prepend() () at /lib/x86_64-linux-gnu/libgtkmm-3.0.so.1
#8  0x00005555556ee45b in (anonymous namespace)::addMessages(Glib::RefPtr<Gtk::ListStore> const&, tr_log_message*) (store=..., head=0x555557b202a0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:361
#9  0x00005555556ee2e4 in MessageLogWindow::Impl::onRefresh() (this=0x555556e343f0) at /home/xxx/dev/transmission/gtk/MessageLogWindow.cc:397
#10 0x00005555556f9412 in sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>::operator()() const (this=0x555556f1d028) at /usr/include/sigc++-2.0/sigc++/functors/mem_fun.h:1991
#11 0x00005555556f9389 in sigc::adaptor_functor<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl> >::operator()() const (this=0x555556f1d020) at /usr/include/sigc++-2.0/sigc++/adaptors/adaptor_trait.h:256
#12 0x00005555556f9361 in sigc::internal::slot_call0<sigc::bound_mem_functor0<bool, MessageLogWindow::Impl>, bool>::call_it(sigc::internal::slot_rep*) (rep=0x555556f1cff0) at /usr/include/sigc++-2.0/sigc++/functors/slot.h:136
#13 0x00007ffff66a5fb2 in  () at /lib/x86_64-linux-gnu/libglibmm-2.4.so.1
#14 0x00007ffff650719a in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#15 0x00007ffff650667f in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#16 0x00007ffff6506a38 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#17 0x00007ffff6506acc in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007ffff67b353d in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#19 0x00005555556c78ce in main(int, char**) (argc=1, argv=0x7fffffffe008) at /home/xxx/dev/transmission/gtk/main.cc:127
```

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

https://github.com/transmission/transmission/commit/3ecf0dfccd627c1a137bd6b5103f9190e0d96af6

### PR DESCRIPTION
**Pull request by ckerr**
Fixes #4181.

Fixes #4233.

This is one of the remaining blockers for [4.0.0-beta.2](https://github.com/transmission/transmission/issues?q=+is%3Aissue+milestone%3A4.0.0-beta.2).

---

OK this was fun. There are at least three bugs here, and two of them were new regressions :1st_place_medal: 

- The new variant_converter code saved log levels as strings, e.g. "info" or "error". This is great since it's more readable, but it doesn't play nicely with how GTK handles its client app preferences. I've reverted this to save as an  int in 47244504d700d5e6749b3317b8e179037aafc92f.

- `tr_sessionGetSettings()` wasn't picking up the current log level because that is kept in libtransmission as a global variable rather than a session variable. :dizzy: 2307b12541fdc8fe001d8b9fce8d96d8c9a05af5 adds a quick patchup to store the global in the target container in `tr_sessionGetSettings()`.

- The GTK client's message log handled the initial filtering in a weird way.  689baed2aa48e79bc1aec725078ead44db31ece0 triers to ensure that (a) _some_ row is selected, even if prefs has an invalid value (e.g. trace-level events) and (b) the log list is refiltered based on whatever row is selected when the window is opened.

CC @Coeur, @mikedld

### CODE REVIEW

**Comment by GaryElshaw** [comment]
> two of them were new regressions 🥇

It wasn't me, i was never there...

**Comment by Coeur** [comment]
If it "_Fixes #4233_", do you mean that we could have kept Trace as a viable option? Or are you removing Trace from UI?

File: gtk/MessageLogWindow.cc [hunk 1]
```diff
@@ -141,13 +142,30 @@ void MessageLogWindow::Impl::scroll_to_bottom()
...
+        {
+            default_row = std::size(items);
+        }
+
         items.emplace_back(name, level);
     }
+
     gtr_combo_box_set_enum(*level_combo, items);
-    gtr_combo_box_set_active_enum(*level_combo, gtr_pref_int_get(TR_KEY_message_level));
+    level_combo->set_active(current_row ? *current_row : default_row);
```
Comment (mikedld) [inline gtk/MessageLogWindow.cc:168]: I don't think breaking away from the `gtr_combo_box_set_active_enum()` abstraction is a good idea; I'd rather change its interface to either accept an [optional] default value and handle it internally
```cpp
gtr_combo_box_set_active_enum(*level_combo, gtr_pref_int_get(TR_KEY_message_level), TR_LOG_INFO);
```
or return a `bool` to for something like
```cpp
if (!gtr_combo_box_set_active_enum(*level_combo, gtr_pref_int_get(TR_KEY_message_level)))
{
    gtr_combo_box_set_active_enum(*level_combo, TR_LOG_INFO);
}
```

If we do break away though,
```suggestion
    level_combo->set_active(current_row.value_or(default_row));
```

**Review by Coeur — APPROVED** [review]

**Comment by ckerr** [comment]
> OK, I see from the code change that you're removing the Trace option.

Yes. This is the GTK followup to the macOS client changes that removed Trace there. :+1:

File: gtk/MessageLogWindow.cc [hunk 2]
```diff
@@ -141,13 +142,30 @@ void MessageLogWindow::Impl::scroll_to_bottom()
...
+
+    auto current_row = std::optional<int>{};
+    auto default_row = int{ 0 };
+
     auto items = std::vector<std::pair<Glib::ustring, int>>{};
     for (auto const& [level, name] : level_names_)
     {
+        if (level == current_level)
+        {
+            current_row = std::size(items);
```
Comment (mikedld) [inline gtk/MessageLogWindow.cc:156]: Here and below: assigning `size_t` to `int` will probably generate a warning.

File: gtk/MessageLogWindow.cc [hunk 3]
```diff
@@ -455,14 +473,11 @@ MessageLogWindow::Impl::Impl(
...
-          { TR_LOG_WARN, C_("Logging level", "Warning") },
-          { TR_LOG_INFO, C_("Logging level", "Information") },
-          { TR_LOG_DEBUG, C_("Logging level", "Debug") },
-          { TR_LOG_TRACE, C_("Logging level", "Trace") },
-      } }
+    , level_names_{ { { TR_LOG_CRITICAL, C_("Logging level", "Critical") },
+                      { TR_LOG_ERROR, C_("Logging level", "Error") },
+                      { TR_LOG_WARN, C_("Logging level", "Warning") },
+                      { TR_LOG_INFO, C_("Logging level", "Information") },
+                      { TR_LOG_DEBUG, C_("Logging level", "Debug") } } }
```
Comment (mikedld) [inline gtk/MessageLogWindow.cc:480]: I usually add trailing comma to avoid such extra and uneven indentation.

File: gtk/MessageLogWindow.cc (hunk 1 above)
Comment (ckerr) [reply gtk/MessageLogWindow.cc:168]: Ah! I always forget about value_or

Comment (ckerr) [reply gtk/MessageLogWindow.cc]: Updated in 7dc51830a91252b0f06a80c51dd78687d58ff7d6. Keeping `gtr_combo_box_set_active_enum()` allows us to drop the size_t / int mismatch too.

File: gtk/MessageLogWindow.cc (hunk 3 above)
Comment (ckerr) [reply gtk/MessageLogWindow.cc]: Trailing comma added in 7dc51830a91252b0f06a80c51dd78687d58ff7d6

Comment (mikedld) [reply gtk/MessageLogWindow.cc]: Slightly misplaced, should be after the element, not the whole value ;)

Comment (ckerr) [reply gtk/MessageLogWindow.cc]: In my defense, I saw the error before reading your comment -- but I appreciate the review either way :)

**Review by mikedld — APPROVED** [review]

**Pull request merged**